### PR TITLE
add windowmasker

### PR DESCRIPTION
**Pull request by bgruening**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by bgruening** [comment]
ping @fubar2

**Comment by fubar2** [comment]
Thanks @bgruening - this will be very useful.

File: tools/windowmasker/windowmasker_mkcounts.xml [hunk 1]
```diff
@@ -0,0 +1,101 @@
+<tool id="windowmasker_mkcounts" name="WindowMasker_mkcounts" version="1.0" profile="22.01">
```
Comment (bgruening) [inline tools/windowmasker/windowmasker_mkcounts.xml:1]:
```suggestion
<tool id="windowmasker_mkcounts" name="WindowMasker mkcounts" version="1.0" profile="22.01">
```

File: tools/windowmasker/.shed.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+name: windowmasker
+owner: iuc
+description: Identify repetitive regions using WindowMasker
+type: unrestricted
+homepage_url: https://www.ncbi.nlm.nih.gov/IEB/ToolBox/CPP_DOC/lxr/source/src/app/winmasker/
+remote_repository_url: https://github.com/galaxyproject/tools-iuc/tree/master/tools/windowmasker/
```
Comment (bgruening) [inline tools/windowmasker/.shed.yml:6]:
```suggestion
remote_repository_url: https://github.com/galaxyproject/tools-iuc/tree/main/tools/windowmasker/
``

File: tools/windowmasker/windowmasker_ustat.xml [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+            </change_format>
+        </data>
+    </outputs>
+    <tests>
+        <test>
+            <!-- Test WindowMasker ustat with fasta output -->
+            <param name="fasta_input" value="contigs.fa" ftype="fasta" />
+            <param name="mkcount_input" value="chr4_part.oascii.counts" ftype="txt" />
+            <param name="output_format" value="fasta" />
+            <output name="mask_output" file="contigs.wm.fa" />
```
[the remark this answers is not in the document]
Comment (bgruening) [reply tools/windowmasker/windowmasker_ustat.xml:128]:
```suggestion
            <output name="mask_output" file="contigs.wm.fa" ftype="fasta" />
```

File: tools/windowmasker/windowmasker_ustat.xml [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+        </test>
+        <test>
+            <!-- Test WindowMasker ustat with ASN.1 text output -->
+            <param name="fasta_input" value="contigs.fa" ftype="fasta" />
+            <param name="mkcount_input" value="chr4_part.oascii.counts" ftype="txt" />
+            <param name="output_format" value="maskinfo_asn1_text" />
+            <output name="mask_output" file="contigs.wm.asn1" />
```
Comment (bgruening) [inline tools/windowmasker/windowmasker_ustat.xml:135]:
```suggestion
            <output name="mask_output" file="contigs.wm.asn1" ftype="maskinfo-asn1"/>
```

File: tools/windowmasker/windowmasker_ustat.xml [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+        </test>
+        <test>
+            <!-- Test WindowMasker ustat with ASN.1 binary output -->
+            <param name="fasta_input" value="contigs.fa" ftype="fasta" />
+            <param name="mkcount_input" value="chr4_part.oascii.counts" ftype="txt" />
+            <param name="output_format" value="maskinfo_asn1_bin" />
+            <output name="mask_output" file="contigs.wm.asnb" />
```
Comment (bgruening) [inline tools/windowmasker/windowmasker_ustat.xml:142]:
```suggestion
            <output name="mask_output" file="contigs.wm.asnb" ftype="maskinfo-asn1-binary"/>
```

File: tools/windowmasker/windowmasker_ustat.xml [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+        </test>
+        <test>
+            <!-- Test WindowMasker ustat with XML output -->
+            <param name="fasta_input" value="contigs.fa" ftype="fasta" />
+            <param name="mkcount_input" value="chr4_part.oascii.counts" ftype="txt" />
+            <param name="output_format" value="maskinfo_xml" />
+            <output name="mask_output" file="contigs.wm.xml" />
```
Comment (bgruening) [inline tools/windowmasker/windowmasker_ustat.xml:149]:
```suggestion
            <output name="mask_output" file="contigs.wm.xml" ftype="xml"/>
```

File: tools/windowmasker/windowmasker_ustat.xml [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+            <param name="fasta_input" value="contigs.fa" ftype="fasta" />
+            <param name="mkcount_input" value="chr4_part.oascii.counts" ftype="txt" />
+            <param name="set_t_high" value="10" />
+            <param name="set_t_low" value="9" />
+            <param name="t_extend" value="5" />
+            <param name="t_low" value="9" />
+            <param name="t_high" value="10" />
+            <param name="t_thres" value="20" />
+            <param name="window" value="50" />
+            <output name="mask_output" file="contigs.advanced_wm.bed" />
```
Comment (bgruening) [inline tools/windowmasker/windowmasker_ustat.xml:162]:
```suggestion
            <output name="mask_output" file="contigs.advanced_wm.bed" ftype="bed"/>
```

File: tools/windowmasker/windowmasker_ustat.xml [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+        </test>
+        <test>
+            <!-- Test WindowMasker ustat without dust -->
+            <param name="fasta_input" value="contigs.fa" ftype="fasta" />
+            <param name="mkcount_input" value="chr4_part.oascii.counts" ftype="txt" />
+            <param name="use_dust_selector" value="no" />
+            <output name="mask_output" file="contigs.nodust_wm.bed" />
```
Comment (bgruening) [inline tools/windowmasker/windowmasker_ustat.xml:169]:
```suggestion
            <output name="mask_output" file="contigs.nodust_wm.bed" ftype="bed"/>
```

File: tools/windowmasker/windowmasker_ustat.xml [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+        </test>
+        <test>
+            <!-- Test WindowMasker ustat with parse Seq-ids -->
+            <param name="fasta_input" value="contigs.fa" ftype="fasta" />
+            <param name="mkcount_input" value="chr4_part.oascii.counts" ftype="txt" />
+            <param name="parse_seqids" value="-parse_seqids" />
+            <output name="mask_output" file="contigs.seqid_wm.bed" />
```
Comment (bgruening) [inline tools/windowmasker/windowmasker_ustat.xml:176]:
```suggestion
            <output name="mask_output" file="contigs.seqid_wm.bed" ftype="bed"/>
```

**Review by bernt-matthias — APPROVED** [review]

**Pull request merged**